mirrord/agent/src/steal/connection.rs

```diff
@@ -190,9 +192,25 @@ impl Client {
                                 request_id,
                                 port: request.port,
                             }));
-                        if tx.send(message).await.is_err() || is_last {
+
+                        if let Err(e) = tx.send(message).await {
+                            warn!(?e, ?connection_id, ?request_id, ?request.port, "failed to send chunked request start");
                             return;
                         }
+                        if is_last {
+                            let message = DaemonTcp::HttpRequestChunked(ChunkedRequest::Body(
+                                ChunkedHttpBody {
+                                    frames: Vec::new(),
+                                    is_last,
+                                    connection_id,
+                                    request_id,
+                                },
+                            ));
+                            if let Err(e) = tx.send(message).await {
+                                warn!(?e, ?connection_id, ?request_id, ?request.port, "failed to send chunked request empty body");
+                                return;
+                            }
+                        }
                     }
                 }
```

Comment:
I think you can just remove this entirely and don't use `is_last` from first `next_frames()` call at all. Same thing will happen in the first iteration of the loop below

Reply:
Smart